src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6045,6 +6114,9 @@ bool canBeNullableType()
             return type;
         }
 
+        private bool IsFunctionPointerStart() => CurrentToken.Kind == SyntaxKind.DelegateKeyword
+                                                 && PeekToken(1).Kind == SyntaxKind.AsteriskToken;
```

Comment:
nit: `=>` on next line indented.  Then no wrapping of the actual expr.

Reply:
Consider `&& peekToken(2).Kind = SyntaxKind.IdentifierToken && peekToken(2).King == SyntaxKind.LessThanToken`.

---
In reply to: [348016475](https://github.com/dotnet/roslyn/pull/39879#discussion_r348016475) [](ancestors = 348016475)

Reply:
`delegate*` is already unambiguous, plus I would need to add handling for the calling convention which would not be trivial. I'm going to leave this as is.

---
In reply to: [348134075](https://github.com/dotnet/roslyn/pull/39879#discussion_r348134075) [](ancestors = 348134075,348016475)